src/main/java/gregtech/common/covers/CoverPump.java

```diff
@@ -75,25 +78,35 @@ public CoverPump(@NotNull CoverDefinition definition, @NotNull CoverableView cov
         this.maxFluidTransferRate = mbPerTick;
         this.transferRate = mbPerTick;
         this.fluidLeftToTransferLastSecond = transferRate;
-        this.fluidFilter = new FluidFilterContainer(this, this::shouldShowTip);
+        this.fluidFilterContainer = new FluidFilterContainer(this);
+    }
+
+    public void setStringTransferRate(String s) {
+        this.fluidFilterContainer.setTransferSize(
+                getBucketMode() == BucketMode.MILLI_BUCKET ?
+                        Integer.parseInt(s) :
```

Comment:
`parseInt` will throw a `NumberFormatException` if the string cannot be parsed to an integer. Is this string guaranteed to be sanitized when this method is utilized? If not, this should use a try-catch block. Also, does the method need to be `public`?

Reply:
string sanitization should be handled by the text widget before this method is called